tests/system/contracts/BridgeOut.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.28;
+
+import {IAssetsBridge} from "./interfaces/IAssetsBridge.sol";
+import {IBTC} from "./interfaces/IBTC.sol";
+import {IERC20} from "./interfaces/solidity/IERC20.sol";
+
+contract SimpleToken {
+    mapping(address => uint256) private _balances;
+    mapping(address => mapping(address => uint256)) private _allowances;
+
+    uint256 private _totalSupply;
+    string public name = "Simple Token";
+    string public symbol = "SIMPLE";
+    uint8 public decimals = 18;
+
+    event Transfer(address indexed from, address indexed to, uint256 value);
+    event Approval(address indexed owner, address indexed spender, uint256 value);
+
+    // View functions for basic functionality
+    function balanceOf(address account) public view returns (uint256) {
+        return _balances[account];
+    }
+
+    function allowance(address owner, address spender) public view returns (uint256) {
+        return _allowances[owner][spender];
+    }
+
+    function totalSupply() public view returns (uint256) {
+        return _totalSupply;
+    }
+
+    // Mint function - creates new tokens
+    function mint(address to, uint256 amount) public {
+        _totalSupply += amount;
+        _balances[to] += amount;
+
+        emit Transfer(address(0), to, amount);
+    }
+
+    // Approve function - allows spender to use owner's tokens
+    function approve(address spender, uint256 amount) public returns (bool) {
+        _allowances[msg.sender][spender] = amount;
+
+        emit Approval(msg.sender, spender, amount);
+        return true;
+    }
+
+    // TransferFrom function - transfers tokens on behalf of another address
+    function transfer(address to, uint256 amount) public returns (bool) {
+        _balances[msg.sender] -= amount;
+        _balances[to] += amount;
+
+        emit Transfer(msg.sender, to, amount);
+        return true;
+    }
+
+    // TransferFrom function - transfers tokens on behalf of another address
+    function transferFrom(address from, address to, uint256 amount) public returns (bool) {
+        _balances[from] -= amount;
+        _balances[to] += amount;
+        _allowances[from][msg.sender] -= amount;
+
+        emit Transfer(from, to, amount);
+        return true;
+    }
+
+    function burnFrom(address from, uint256 amount) public {
+        require(from != address(0), "ERC20: burn from zero address");
+        require(_balances[from] >= amount, "ERC20: burn amount exceeds balance");
+        require(_allowances[from][msg.sender] >= amount, "ERC20: burn amount exceeds allowance");
+
+        _balances[from] -= amount;
+        _totalSupply -= amount;
+        _allowances[from][msg.sender] -= amount;
+
+        emit Transfer(from, address(0), amount);
+    }
+}
+
+/// @title BridgeOut
+/// @notice Handles various bridgeOut scenarios for Mezo native asset bridge.
+contract BridgeOut {
```

Comment:
WDYT about naming this `BridgeOutDelegate`? The current name may suggest this contract handles bridging while it doesn't.